server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamTests.java

```diff
@@ -1146,11 +1147,34 @@ public void testGetIndicesPastRetention() {
             );
             Metadata metadata = builder.build();
 
-            assertThat(dataStream.getIndicesPastRetention(metadata::index, () -> now).isEmpty(), is(true));
+            assertThat(dataStream.getIndicesPastRetention(metadata::index, () -> now, randomGlobalRetention()).isEmpty(), is(true));
         }
 
         {
-            // no retention configured so we expect an empty list
+            // no retention configured but we have default retention
+            DataStreamGlobalRetention globalRetention = new DataStreamGlobalRetention(
+                TimeValue.timeValueMillis(2500),
+                randomBoolean() ? TimeValue.timeValueMillis(randomIntBetween(2500, 5000)) : null
+            );
+            Metadata.Builder builder = Metadata.builder();
+            DataStream dataStream = createDataStream(
+                builder,
+                dataStreamName,
+                creationAndRolloverTimes,
+                settings(IndexVersion.current()),
+                new DataStreamLifecycle()
+            );
+            Metadata metadata = builder.build();
+
+            List<Index> backingIndices = dataStream.getIndicesPastRetention(metadata::index, () -> now, globalRetention);
+            assertThat(backingIndices.size(), is(2));
+            assertThat(backingIndices.get(0).getName(), is(DataStream.getDefaultBackingIndexName(dataStreamName, 1)));
```

Comment:
Might be better to do `dataStream.getIndices().get(0).getName()` and `dataStream.getIndices().get(1).getName()`. I've had a lot of problems with this method in tests that were run across midnight (extremely unlikely in a unit test like this, but easier to deal with it now than later).

Reply:
Ah yes! Great point!

Reply:
I changed it on both retention related tests